### ISSUE TITLE
Add sats amount when a fixed amount order is created

### ISSUE THREAD

**Issue by grunch** [MostroP2P/mostro-web#22]
When a new buy order with fixed amount is created, the order that will be send to Mostro should have the same amount from the lightning invoice.

```json
{
  "Order": {
    "version": 1,
    "pubkey": "0000147e939bef2b81c27af4c1b702c90c3843f7212a34934bff1e049b7f1427",
    "action": "NewOrder",
    "content": {
      "Order": {
        "kind": "Buy",
        "status": "Pending",
        "amount": 123,
        "fiat_code": "VES",
        "fiat_amount": 100,
        "payment_method": "face to face",
        "premium": 1,
        "buyer_invoice": "lnbc1230n1pja2q4jpp5e53c8hpnjrf4297ld85pv2ka825jgr632zatse50c7jxu0xpducsdqqcqzzsxqyz5vqsp50pnuc8hha6fkwk6ademx0v9468rknmsdp4u07snfrxv947z6lw8s9qyyssqz6hc8g0gm8n39p5ny2f73az06ah9xm3t83k3q8s0nuku46c0gm4kv7r68dn6ncvnmfrrtauxhjy8u6pjxn5jw65wv03nsxtmu52pawsp83a7lk",
        "created_at": 0
      }
    }
  }
}
```

If the user uses a lightning address we can't have a fixed amount, so in this case `amount = 0`:

```json
{
  "Order": {
    "version": 1,
    "pubkey": "0000147e939bef2b81c27af4c1b702c90c3843f7212a34934bff1e049b7f1427",
    "action": "NewOrder",
    "content": {
      "Order": {
        "kind": "Buy",
        "status": "Pending",
        "amount": 0,
        "fiat_code": "VES",
        "fiat_amount": 100,
        "payment_method": "face to face",
        "premium": 1,
        "buyer_invoice": "mostro_p2p@ln.tips",
        "created_at": 0
      }
    }
  }
}
```

For better UX, we should have a `input` for amount like we have in new sell, this allows to the user to add the amount with or without LN invoice, if the user wants to create a buy order with fixed amount but without sending the ln invoice at the moment we should allow it, the order should be like this:

```json
{
  "Order": {
    "version": 1,
    "pubkey": "0000147e939bef2b81c27af4c1b702c90c3843f7212a34934bff1e049b7f1427",
    "action": "NewOrder",
    "content": {
      "Order": {
        "kind": "Buy",
        "status": "Pending",
        "amount": 0,
        "fiat_code": "VES",
        "fiat_amount": 100,
        "payment_method": "face to face",
        "premium": 1,
        "created_at": 0
      }
    }
  }
}
```

If the user add a LN invoice but didn't add the amount, the client should be smart enough to get the amount from the LN invoice and include it on amount.

**Comment by bilthon** [comment]
I agree that an input field for the amount should be introduced. However when you say:

> If the user uses a lightning address we can't have a fixed amount, so in this case amount = 0

I don't get it, if the user gives us a lightning address (or an invoice without an amount for that matter) that's precisely when amount should be different than zero, otherwise this will be considered a market price order. From the [docs](https://mostro.network/messages/new_order.html):

> amount: 0 for when we want to sell with at market price, otherwise the amount in satoshis

And finally:

> If the user add a LN invoice but didn't add the amount, the client should be smart enough to get the amount from the LN invoice and include it on amount.

This makes sense, we should maybe also reset & disable the input field when an invoice with an amount is pasted as to override it with the amount we detect in the invoice, what do you think?

**Comment by grunch** [comment]
you are right, lightning addresses and invoices without amount can be used with a fixed amount too

**Comment by bilthon** [comment]
Fixed in 0c284270687438ecd4c9e4be5ddadcf4738e93d1
